src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -1586,20 +1586,17 @@ public function isNoCache()
      */
     public function getPreferredFormat(?string $default = 'html'): ?string
     {
-        if (null !== $this->preferredFormat) {
+        if (null !== $this->preferredFormat || null !== $this->preferredFormat = $this->getRequestFormat(null)) {
             return $this->preferredFormat;
         }
 
-        $preferredFormat = null;
-        foreach ($this->getAcceptableContentTypes() as $contentType) {
-            if ($preferredFormat = $this->getFormat($contentType)) {
-                break;
+        foreach ($this->getAcceptableContentTypes() as $mimeType) {
+            if ($this->preferredFormat = $this->getFormat($mimeType)) {
+                return $this->preferredFormat;
```

Comment:
this is not going through `getRequestFormat` anymore. Is it expected ?

Reply:
See the first diff, it's calling to `getRequestFormat`.

Reply:
ah indeed. The argument passed to `getRequestFormat` is the fallback value when there is no `_format` attribute.